### PR TITLE
Add trajectory preview subtree for convenience

### PR DESCRIPTION
**Pull request by pac48**
This PR address https://github.com/PickNikRobotics/moveit_studio/issues/3545
I added a new objective to the `picknik_ur_base_config` package to preview a trajectory if the user is available.

### CODE REVIEW

File: src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml
Comment (sea-bass) [inline src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml:8]:
```suggestion
      <Action ID="WaitForUserTrajectoryApproval" solution="{solution}"/>

File: src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml
Comment (sea-bass) [inline src/picknik_ur_base_config/objectives/wait_for_trajectory_approval_if_user_available.xml:9]: We actually use the complete syntax here.

```suggestion
    <Control ID="Fallback" name="wait_for_approval_if_user_available">
      <Decorator ID="Inverter">
        <Action ID="IsUserAvailable"/>
      </Decorator>
      <Action ID="WaitForUserTrajectoryApproval" solution="{mtc_solution}"/>
    </Control>
```

**Review by sea-bass — CHANGES_REQUESTED** [review]
The subtree looks great -- but I think we should also use it as a subtree in all the Objectives that currently use it!

For example, in [this objective](https://github.com/PickNikRobotics/moveit_studio_ws/blob/main/src/picknik_ur_site_config/objectives/pick_object.xml#L15-L20).

old:
```
            <Fallback name="wait_for_approval_if_user_available">
                <Inverter>
                    <Action ID="IsUserAvailable"/>
                </Inverter>
                <Action ID="WaitForUserTrajectoryApproval" solution="{pick_object_solution}"/>
            </Fallback>
            <Action ID="ExecuteMTCTask" solution="{pick_object_solution}"/>
```

new:
```
            <SubTree ID="Wait for Trajectory Approval if User Available" solution="{mtc_solution}">
```

It should end up looking something like this:

image

**Comment by pac48** [comment]
> The subtree looks great -- but I think we should also use it as a subtree in all the Objectives that currently use it!
> 
> For example, in [this objective](https://github.com/PickNikRobotics/moveit_studio_ws/blob/main/src/picknik_ur_site_config/objectives/pick_object.xml#L15-L20).
> 
> old:
> 
> ```
>             <Fallback name="wait_for_approval_if_user_available">
>                 <Inverter>
>                     <Action ID="IsUserAvailable"/>
>                 </Inverter>
>                 <Action ID="WaitForUserTrajectoryApproval" solution="{pick_object_solution}"/>
>             </Fallback>
>             <Action ID="ExecuteMTCTask" solution="{pick_object_solution}"/>
> ```
> 
> new:
> 
> ```
>             <SubTree ID="Wait for Trajectory Approval if User Available" solution="{mtc_solution}">
> ```
> 
> It should end up looking something like this:
> 
> image

Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
```            
<SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
```

**Comment by sea-bass** [comment]
> Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
> 
> ```
> <SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
> ```

Yes, good call, but it should be `"{pick_object_solution}"` based on what I see.

**Comment by pac48** [comment]
> > Should the new block be like this instead? I don't see where {mtc_solution} would be referenced from.
> > ```
> > <SubTree ID="Wait for Trajectory Approval if User Available" solution="{pick_object_task}">
> > ```
> 
> Yes, good call, but it should be `"{pick_object_solution}"` based on what I see.

Great, I added `"pick_object_solution"` and fixed the rest.

**Review by adlarkin — APPROVED** [review]

**Pull request merged**
